pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go

```diff
@@ -179,15 +179,35 @@ KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/k
 .PHONY: kustomize
 kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
 $(KUSTOMIZE): $(LOCALBIN)
+ifeq "$(uname -s)" "Darwin"
```

Comment:
It is hard to keep maintained and not sustainable ( Spaghetti Code ) 
Also, we cannot do that. We cannot add a script from your repo in the default scaffold. 

**About etcd bin**

The etcd bin is used for ENV TEST and is shipped via the kubebuilder-tools, see: 
- https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases 
- https://storage.googleapis.com/kubebuilder-tools

Also, we are close to having that bin from the project:
- https://github.com/etcd-io/etcd/issues/14001#issuecomment-1146800339

As soon we have the binary there we can trigger the job in the cloud to build the kubebuilder-tools for darwin/arm64. To better understand it check the issue: https://github.com/kubernetes-sigs/kubebuilder/issues/2664

**About kustomize bin**

You can follow up: https://github.com/kubernetes-sigs/kustomize/issues/4612
Users will be able to use it with kustomize/v2 and go/v4-alpha, see: https://github.com/kubernetes-sigs/kubebuilder/pull/2552

I am not sure if we should assume the decomposability to build kustomize v2 for darwin/arm64 to make it work with go/v3 stable. That will add this kind of complexity to the Makefile and also we would need to properly build it in the cloud as we do with the kubebuilder-tools instead. It seems out of the scope from Kubebuilder itself.

However, we could doc the script for example to allow users to build it by themselves and doc as an option. But the script also needs to leave in the repo or some official place and not in our fork. 

To know more about what is required to support apple silicon and what was done or not so far see: https://github.com/kubernetes-sigs/kubebuilder/issues/1932

Reply:
Sounds good. I think at least adding some docs for a workaround would be good so that users can begin using Kubebuilder/Operator SDK to develop operators on apple silicon. 

Just as an aside, the goal of this PR is to add those scripts to Kubebuilder as well so that the scripts that are run in the makefile are from the official Kubebuilder repo, I just left it pointing to my fork so we could test that it actually works since Kubebuilder doesn't have the scripts in place yet and would modify it to point to Kubebuilder prior to merging.

If this isn't a direction that we want to explore that is fine and I can close this PR, just thought I would give it a shot at trying to provide a solution that would allow development with go/v3 on Apple Silicon. Thanks for the feedback!

Reply:
> Sounds good. I think at least adding some docs for a workaround would be good so that users can begin using Kubebuilder/Operator SDK to develop operators on apple silicon.

- For the Kubebuilder side we can begin to support Apple Silicon from go/v4-alpha and with kustomize/v2-alpha
- For SDK we need to discuss how we will do it there? Could we accept the breaking change for ansible/helm/hybrid instead of creating a new plugin or not for example?

> Just as an aside, the goal of this PR is to add those scripts to Kubebuilder as well so that the scripts that are run in the makefile are from the official Kubebuilder repo, I just left it pointing to my fork so we could test that it actually works since Kubebuilder doesn't have the scripts in place yet and would modify it to point to Kubebuilder prior to merging.

I think we have some options to address these ones, follow some ideas/suggestion:

**Regards the bins used in kubebuilder-tools/env test:**

a) We could Improve the kubebuilder-tools scripts for if the bin does not exist we build them for the arch informed so that it can work in the cloud and can be used by the authors. Also, we can improve the doc: https://book.kubebuilder.io/reference/artifacts.html and let the users know about that and that they can use it to build for any platform 

**Regards kustomize**

- For the next release, we will need to clarify its support with go/v4-alpha in the quickstart probably, I am looking if we can merge https://github.com/kubernetes-sigs/kubebuilder/pull/2552 so that we can work on this
- Also, we could add a new section FAQ as we have in SDK docs ( see that we have many issues track to do that and add the info there ) then, we can have a directory under it called script and add the script that will let users know that they can use to build kustomize v2 for apple silicon or for any other platform that they need. 

**Regards SDK**
After we add the changes here in SDK we can use/ link and redirect users to not need to duplicate the content and be required to keep maintained two places

WDYT? Is that make sense?
Could you shape your scripts so we provide them to the users via the FAQ? Would you like to contribute with the kubebuilder-tools scripts to make them be able to build the project when the bin is not found?

Reply:
Sure, if you wish we can track issues with that as well. 
and by sure .. feel free to ping if you need any help